src/backend/mpi/swdsm.cpp

```diff
@@ -769,17 +769,20 @@ void * loadcacheline(void * x){
 
 		unsigned long offset = getOffset(lineAddr);
 		if(isPowerOf2((tempsharer)&invid) && tempsharer != id && prevsharer == 0){ //Other private. but may not have loaded page yet.
-			long ownid = (long)(tempsharer&invid);
-			unsigned long owner;
-			for(n=0; n<numtasks; n++){
-				if((unsigned)(1<<n)==(unsigned)ownid){
+			unsigned long ownid = tempsharer&invid; // remove own bit
+			static const unsigned long invalid_owner = static_cast<unsigned long>(-1);
```

Comment:
you are correct about the static const, should be constexpr.
static_cast is used to document that this is an invalid value for the type in question, I have not checked whether any or all compilers accept this silently.

However, I am not sure that creating this at the global scope is helpful or harmful to readability of the code. If used at global code it definitely should be static to avoid making it part of the ABI, right?